halyard-deploy/src/main/resources/kubernetes/manifests/deployment.yml

```diff
@@ -19,6 +19,14 @@ spec:
     matchLabels:
       app: spin
       cluster: spin-{{ name }}
+  {% if deploymentStrategy != null %}
+  strategy:
+    {{ deploymentStrategy.type }}:
```

Comment:
It might be good to be a bit more guarded here and only inject this block if `type == rollingUpdate`. If you switch from `rollingUpdate` to `recreate` but leave the `strategySettings` in place it could result in an invalid manifest.

Reply:
+1

Reply:
Sounds good. I thought about this too, but the only reason I didn't limit it to rollingUpdate explicitly was to try to give room to support other update types in the future, but then it would have to change anyways because of the spelling/casing difference below. 
I'll make this change and push it up.

Reply:
The length check is necessary so that someone changes from recreate back to rollingUpdate and doesn't specify any settings it won't create a bad manifest.